src/uu/cksum/src/cksum.rs

```diff
@@ -116,23 +156,23 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
 
     let matches = uu_app().try_get_matches_from(args)?;
 
-    let files: Vec<String> = match matches.get_many::<String>(options::FILE) {
-        Some(v) => v.clone().map(|v| v.to_owned()).collect(),
-        None => vec![],
+    let algo_name: &str = match matches.get_one::<String>(options::ALGORITHM) {
+        Some(v) => v,
+        None => "crc",
     };
 
-    if files.is_empty() {
-        let (crc, size) = cksum("-")?;
-        println!("{crc} {size}");
-        return Ok(());
-    }
+    let (name, algo, bits) = detect_algo(algo_name);
+    let opts = Options {
+        algo_name: name,
+        digest: algo,
+        output_bits: bits,
+    };
+
+    match matches.get_many::<String>(options::FILE) {
```

Comment:
May I ask more about which function in detailed?

Reply:
Something like `matches.get_many::<OsString>(options::FILE)`. But we can leave it for now, because it's not part of the change you made. So we'll do this some other time.